builder/oci/config.go

```diff
@@ -25,8 +26,13 @@ import (
 
 type CreateVNICDetails struct {
 	// fields that can be specified under "create_vnic_details"
-	AssignPublicIp      *bool                             `mapstructure:"assign_public_ip" required:"false"`
-	DefinedTags         map[string]map[string]interface{} `mapstructure:"defined_tags" required:"false"`
+	AssignPublicIp *bool `mapstructure:"assign_public_ip" required:"false"`
+	// HCL cannot be decoded into an interface so for HCL templates you must use the DefinedTagsJson option,
+	// To be used with https://www.packer.io/docs/templates/hcl_templates/functions/encoding/jsonencode
+	// ref: https://github.com/hashicorp/hcl/issues/291#issuecomment-496347585
+	DefinedTagsJson string `mapstructure:"defined_tags_json" required:"false"`
```

Comment:
Is this comment going to end up in the docs ? Should we add something a bit more packer.io docs friendly ?

Reply:
Ah, looking at docs/builders/oci.mdx, looks like no !